pkg/util/openstack/loadbalancer.go

```diff
@@ -169,15 +170,52 @@ func CreateListener(client *gophercloud.ServiceClient, lbID uint64, opts listene
 
 // GetLoadbalancerByName get the load balancer which is in valid status by the given name.
 func GetLoadBalancersByName(client *gophercloud.ServiceClient, name string) ([]loadbalancers.LoadBalancer, error) {
-	var validLBs []loadbalancers.LoadBalancer
+	var allLoadbalancers, validLBs []loadbalancers.LoadBalancer
+	var listOpts loadbalancerv2.ListOpts
+	var oldMarker string
+
+	// Query all the LBs by using the markers.
+	// Set last LB ID as marker.
+	for {
+		var tempLbs []loadbalancers.LoadBalancer
+		var err error
+
+		err = loadbalancers.List(client, listOpts).EachPage(func(page pagination.Page) (bool, error) {
+			tempLbs, err = loadbalancers.ExtractLoadBalancers(page)
+			if err != nil {
+				return false, err
```

Comment:
Shouldn't this be returning nil? Instead of false? Is it the same when it comes to golang?

Reply:
EachPage is a wrapper function to the page function we defined and EachPage function expects bool value to process the next page. We are sending false because as there is no next page to process.